include/onnxruntime/core/graph/graph.h

```diff
@@ -675,7 +676,15 @@ class Graph {
   Note: This currently has linear time complexity. There is room for improvement but it would likely require changes to
   how initializer tensors are stored and tracked.
   */
-  common::Status ReplaceInitializedTensor(const ONNX_NAMESPACE::TensorProto& new_initializer);
+  common::Status ReplaceInitializedTensor(ONNX_NAMESPACE::TensorProto new_initializer);
+
+#if !defined(DISABLE_EXTERNAL_INITIALIZERS)
+  /** This function takes externally provided data for initializers with external data
+  *    and replaces graph initializers with its content.
+  */
+  common::Status InjectExternalInitializedTensors(InlinedHashMap<std::string, OrtValue> external_initializers);
```

Comment:
>InlinedHashMap<std::string, OrtValue> external_initializers [](http://example.com/codeflow?start=50&length=59)

const &?

Reply:
It is intentional by value, see the usage of it.